src/Plivo/Resource/ResourceInterface.cs

```diff
@@ -175,8 +175,12 @@ public async Task<T> DeleteResource<T> (string id, Dictionary<string, object> da
                 return func ();
             } catch (AggregateException ex) {
                 ex.Flatten ();
+                if (ex.InnerExceptions[0] is Newtonsoft.Json.JsonReaderException){
```

Comment:
why cant you throw plivo api error here? rather than generic error, like in case 503 or 502 or something

Reply:
The json exception is raised before we process the response. So this exception willl be raised everytime the response from an api is not in json format

Reply:
Yep, even I found the same thing while testing, Got html error - <title>503 Service Temporarily Unavailable</title>. But the SDK output was shown as the generic error.

Reply:
@renoldthomas-plivo since the error message is not in json, we would get the generic error. Please go through my previous comment once.